### PR TITLE
Feature/tao 8246/add form components/widget hiddenbox

### PR DESCRIPTION
**Pull request by jsconan**
Related to: https://oat-sa.atlassian.net/browse/TAO-8246

Requires: 
 - [x] https://github.com/oat-sa/tao-core-ui-fe/pull/5
 - [x] https://github.com/oat-sa/tao-core-ui-fe/pull/6

This PR is part of a bigger feature set, and is a split of https://github.com/oat-sa/tao-core/pull/2131.
The main purpose is to add a collection of form components.

So here's what we have here:
- `ui/form/widget/hiddenBox` is a double password input widget

Unit tests:
```
npm i
npm run test form
```

Each unit test also comes with a visual playground so you can play with each component to see how they behave. To interact with them:
```
npm run test:keepAlive
```
Then open a browser on http://127.0.0.1:8082/test/form/widget/hiddenBox/test.html

### CODE REVIEW

File: test/form/widget/hiddenBox/test.js [hunk 1]
```diff
@@ -0,0 +1,821 @@
...
+            .on('init', function () {
+                assert.equal(this, instance, 'The instance has been initialized');
+            })
+            .on('ready', function () {
+                assert.equal($container.children().length, 2, 'The container contains an element');
+                ready();
+            })
+            .on('change', function (value, uri) {
+                this.validate();
+                $outputChange.val('value of [' + uri + '] changed to "' + value + '"\n' + $outputChange.val());
```
Comment (ricproenca-tao) [inline test/form/widget/hiddenBox/test.js]: Consider replace `value` for `JSON.stringify(value)` once value is an object.

Comment (jsconan) [reply test/form/widget/hiddenBox/test.js]: Done

File: src/form/widget/providers/hiddenBox.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    /**
+     * Initialize the widget.
+     * @param {hiddenBoxConfig} config
+     */
+    init(config) {
+        // the type will be reflected to the HTML markup
+        config.widgetType = 'hidden-box';
+
+        // config for the confirmation field
+        config.confirmation = _.defaults(config.confirmation || {}, {
```
Comment (krampstudio) [inline src/form/widget/providers/hiddenBox.js]: Maybe the confirmation needs to be optional.
It can be also for a future improvement

Comment (jsconan) [reply src/form/widget/providers/hiddenBox.js]: Do you mean the widget could be either single password input or confirmed password input?
Should not it be better then to have another component that only supplies a single password?

**Pull request merged**